Libraries/dnYara.Interop/Interops/Types/YR_RULES.cs

```diff
@@ -4,41 +4,48 @@
 
 namespace dnYara.Interop
 {
-    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
-    public struct YR_RULES
+    [StructLayout(LayoutKind.Explicit, Size = 64)]
+    public unsafe struct OpaquePthreadMutexT
```

Comment:
This type is one of two types in reality:

* On windows it's a `HANDLE`, aka a `VOID*`, so it's either 32 or 64 bits depending on your OS bit-tedness.
* On unit's it's a `pthread_mutex_t`, which is iirc 40 bits.

This struct consumes the space taken up by the `YR_MUTEX` struct embedded in `YR_RULES` so that the remaining offsets are correct.

Reply:
Nice !! Thanks !